app/src/main/java/org/schabi/newpipe/info_list/LivePreviewDispatcher.kt

```diff
@@ -0,0 +1,95 @@
+package org.schabi.newpipe.info_list
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.TransitionDrawable
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.view.animation.AlphaAnimation
+import android.view.animation.Animation
+import android.widget.ImageView
+import androidx.preference.PreferenceManager
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.functions.BiFunction
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.schabi.newpipe.R
+import org.schabi.newpipe.extractor.stream.StreamInfoItem
+import org.schabi.newpipe.util.RxStreamFrames
+import java.lang.ref.WeakReference
+import java.util.concurrent.TimeUnit
+
+class LivePreviewDispatcher(context: Context) {
+
+    private val preferences = PreferenceManager.getDefaultSharedPreferences(context)
+    private val preferenceKey = context.getString(R.string.show_live_preview_key)
+
+    private var activePreview: Disposable? = null
+    private var currentView: WeakReference<ImageView>? = null
+
+    fun show(target: ImageView, stream: StreamInfoItem) {
+        activePreview?.dispose()
+        if (!preferences.getBoolean(preferenceKey, true)) {
+            return
+        }
+        val last = target.drawable
```

Comment:
`last` is not a meaningful name for this variable. I'd call it `originalThumbnail`